### PR TITLE
`General`: Use details component in course, lectures, exams and tutorial groups views

### PR DESCRIPTION
**Pull request by milljoniaer**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context
This PR adds the added components for details from #7632 into the other entities than exercises. Hence, the look of the Details can be unified there.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Lecture
- 1 Tutorial Group
- 1 Exam

1. Visit Course Detail View
2. Visit Exam Detail View
3. Visit Tutorial Detail View
4. Visit Lecture Detail View
5. Verify that it looks fine and shows the relevant information

#### Performance Review
- [x] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

#### Client

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [course-detail.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5840/latest/artifact/shared/Coverage-Report-Client-Tests/app/course/manage/detail/course-detail.component.ts.html) | 89.47% | ✅ |
| [tutorial-group-detail.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5840/latest/artifact/shared/Coverage-Report-Client-Tests/app/course/tutorial-groups/shared/tutorial-group-detail/tutorial-group-detail.component.ts.html) | 94.28% | ✅|
| [detail-overview-list.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5840/latest/artifact/shared/Coverage-Report-Client-Tests/app/detail-overview-list/detail-overview-list.component.ts.html) | 100% | ✅ |
| [exam-detail.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5840/latest/artifact/shared/Coverage-Report-Client-Tests/app/exam/manage/exams/exam-detail.component.ts.html) | 94.02% | ✅|
| [file-upload-exercise-detail.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5840/latest/artifact/shared/Coverage-Report-Client-Tests/app/exercises/file-upload/manage/file-upload-exercise-detail.component.ts.html) | 97.91% | ✅|
| [modeling-exercise-detail.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5840/latest/artifact/shared/Coverage-Report-Client-Tests/app/exercises/modeling/manage/modeling-exercise-detail.component.ts.html) | 93.05% | ✅|
| [programming-exercise-detail.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5840/latest/artifact/shared/Coverage-Report-Client-Tests/app/exercises/programming/manage/programming-exercise-detail.component.ts.html) | 87.44% | ✅|
| [quiz-exercise-detail.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5840/latest/artifact/shared/Coverage-Report-Client-Tests/app/exercises/quiz/manage/quiz-exercise-detail.component.ts.html) | 100% | ✅ |
| [quiz-exercise-lifecycle-buttons.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5840/latest/artifact/shared/Coverage-Report-Client-Tests/app/exercises/quiz/manage/quiz-exercise-lifecycle-buttons.component.ts.html) | 90.38% | ✅|
| [quiz-exercise-manage-buttons.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5840/latest/artifact/shared/Coverage-Report-Client-Tests/app/exercises/quiz/manage/quiz-exercise-manage-buttons.component.ts.html) | 90% | ✅|
| [quiz-exercise-update.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5840/latest/artifact/shared/Coverage-Report-Client-Tests/app/exercises/quiz/manage/quiz-exercise-update.component.ts.html) | 92.04% | ✅ |
| [quiz-exercise.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5840/latest/artifact/shared/Coverage-Report-Client-Tests/app/exercises/quiz/manage/quiz-exercise.component.ts.html) | 91.3% | ✅|
| [quiz-point-statistic.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5840/latest/artifact/shared/Coverage-Report-Client-Tests/app/exercises/quiz/manage/statistics/quiz-point-statistic/quiz-point-statistic.component.ts.html) | 93.1% | ✅  |
| [feedback.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5840/latest/artifact/shared/Coverage-Report-Client-Tests/app/exercises/shared/feedback/feedback.component.ts.html) | 95% | ✅ |
| [result.service.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5840/latest/artifact/shared/Coverage-Report-Client-Tests/app/exercises/shared/result/result.service.ts.html) | 90.29% | ✅|
| [result.utils.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5840/latest/artifact/shared/Coverage-Report-Client-Tests/app/exercises/shared/result/result.utils.ts.html) | 90.41% | ✅|
| [utils.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5840/latest/artifact/shared/Coverage-Report-Client-Tests/app/exercises/shared/utils.ts.html) | 94.11% | ✅ |
| [text-exercise-detail.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5840/latest/artifact/shared/Coverage-Report-Client-Tests/app/exercises/text/manage/text-exercise/text-exercise-detail.component.ts.html) | 100% | ✅ |
| [lecture-detail.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5840/latest/artifact/shared/Coverage-Report-Client-Tests/app/lecture/lecture-detail.component.ts.html) | 100% | ✅ |
| [detail-overview-navigation-bar.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5840/latest/artifact/shared/Coverage-Report-Client-Tests/app/shared/detail-overview-navigation-bar/detail-overview-navigation-bar.component.ts.html) | 100% | ✅ |

### Screenshots
image
image
image

## Summary by CodeRabbit

- **New Features**
  - Integrated a new `Detail Overview` component across various sections such as courses, exams, lectures, and tutorial groups to enhance the presentation of details.
  - Added a new `Artemis Duration From Seconds` pipe in the exam management module for better time display.

- **Enhancements**
  - Improved the `Tutorial Group Detail` view with a structured presentation and new methods to format session times.
  - Enhanced the `Programming Exercise Detail` component to include additional settings information when applicable.

- **Bug Fixes**
  - Updated Cypress test selectors to reflect changes in the UI, ensuring tests remain accurate and reliable.
  - Fixed the display logic in the `Detail Overview List` component by adding conditional checks and modifying element attributes.

- **Documentation**
  - Updated Cypress E2E test documentation to reflect changes in utility function behavior and page object selectors.

- **Refactor**
  - Streamlined the initialization process of components by making `ngOnInit` asynchronous and refactoring content.
  - Replaced detailed lists with `Detail Overview` components to centralize the presentation of information.

- **Tests**
  - Enhanced existing test cases and added new ones to cover the updated functionality and components, including mock implementations for better isolation.

- **Chores**
  - Updated import statements and module dependencies to align with the new feature integrations.

### CODE REVIEW

**Review by sarpsahinalp — DISMISSED** [review]
Tested on ts2. Worked as expected

**Review by AntonGeTUM — COMMENTED** [review]
Tested on ts11, detail pages look good
For programming exercises, shouldn't the second link say "Show solution submissions" instead of "Show Template submissions" twice?
image

**Review by xHadie — DISMISSED** [review]
_Manually tested on TS11_Legacy_
Everything is looking fine for me, did not notice any degradations!

**Review by nityanandaz — COMMENTED** [review]
Code review: looks neat, except client-style checks :)

**Review by Predixx — DISMISSED** [review]
tested on ts2. Looks like the screenshots

File: src/main/webapp/app/course/tutorial-groups/shared/tutorial-group-detail/tutorial-group-detail.component.html [hunk 1]
```diff
@@ -5,182 +5,18 @@ <h2>{{ tutorialGroup.title }}</h2>
...
-            }
-        </dd>
-    </dl>
+    @if (tutorialDetailSections) {
+        <jhi-detail-overview-list [sections]="tutorialDetailSections"></jhi-detail-overview-list>
+    }
+    <div class="mt-1">
+        <strong>{{ 'artemisApp.entities.tutorialGroup.sessions' | artemisTranslate }}</strong>
+    </div>
+    @if (sessions && sessions.length > 0) {
```
Comment (rabeatwork) [inline src/main/webapp/app/course/tutorial-groups/shared/tutorial-group-detail/tutorial-group-detail.component.html:14]: @if (sessions && sessions.length)

File: src/main/webapp/app/course/tutorial-groups/shared/tutorial-group-detail/tutorial-group-detail.component.ts [hunk 2]
```diff
@@ -54,5 +56,75 @@ export class TutorialGroupDetailComponent implements OnChanges {
                 this.changeDetectorRef.detectChanges();
             }
         }
+        this.getTutorialDetailSections();
+    }
+
+    getTutorialTimeSlotString() {
```
Comment (rabeatwork) [inline src/main/webapp/app/course/tutorial-groups/shared/tutorial-group-detail/tutorial-group-detail.component.ts:62]: Add return type

File: src/main/webapp/app/course/manage/detail/course-detail.component.html [hunk 3]
```diff
@@ -49,252 +49,9 @@
...
-                        <dt><span jhiTranslate="artemisApp.iris.settings.subSettings.enabled.codeEditor">Iris CodeEditor</span></dt>
-                        <dd>
-                            <jhi-iris-enabled [course]="course" [irisSubSettingsType]="CODE_EDITOR" [disabled]="!isAdmin"></jhi-iris-enabled>
-                        </dd>
-                    </div>
-                }
-            </dl>
         </div>
     }
+    @if (course) {
```
Comment (rabeatwork) [inline src/main/webapp/app/course/manage/detail/course-detail.component.html:54]: is @if (course) correct or should it be  @if (courseDetailSections)?

File: src/main/webapp/app/exam/manage/exams/exam-detail.component.html [hunk 4]
```diff
@@ -108,163 +108,7 @@ <h2 id="exam-detail-title">
...
-                        <span jhiTranslate="artemisApp.exam.notSet"></span>
-                    }
-                    <dt><span jhiTranslate="artemisApp.examManagement.confirmationEndText">Exam end confirmation text</span></dt>
-                    @if (formattedConfirmationEndText) {
-                        <dd id="exam-confirmation-end-text" class="editor-outline-background" [innerHTML]="formattedConfirmationEndText"></dd>
-                    } @else {
-                        <span jhiTranslate="artemisApp.exam.notSet"></span>
-                    }
-                </dl>
+                <jhi-detail-overview-list [sections]="examDetailSections"></jhi-detail-overview-list>
```
Comment (rabeatwork) [inline src/main/webapp/app/exam/manage/exams/exam-detail.component.html:111]: Do you need to check for examDetailSections, don't you? (@if (examDetailSections))

**Review by konrad2002 — APPROVED** [review]
manually tested on ts5, could not find any issues with the details

**Review by mateusmm01 — APPROVED** [review]
Demonstrated in testing session. LGTM!

**Review by reschandreas — APPROVED** [review]
demonstrated in testing session, works as expected, lgtm

**Review by laurenzfb — APPROVED** [review]
demonstrated in testing session, code also lgtm

**Review by reschandreas — APPROVED** [review]
reapprove

**Review by nityanandaz — COMMENTED** [review]
A tiny typo

**Review by mateusmm01 — DISMISSED** [review]
Reaprove

**Review by laurenzfb — APPROVED** [review]
reapprove after changes

**Review by reschandreas — DISMISSED** [review]
reapprove

**Review by mateusmm01 — DISMISSED** [review]
Reapprove after code changes

**Review by nityanandaz — APPROVED** [review]
Re-approve code

**Review by Predixx — DISMISSED** [review]
Reapproved

**Review by rabeatwork — APPROVED** [review]

**Review by georgyia — DISMISSED** [review]
manually tested on ts2, looks good, the information is relevant

**Review by georgyia — DISMISSED** [review]
reapprove

**Review by rstief — APPROVED** [review]
Tested on ts4, looks good 👍

**Review by reschandreas — APPROVED** [review]
tested in testing session, works as expected, thanks!

**Review by laurenzfb — APPROVED** [review]
tested on ts4, works

**Review by georgyia — APPROVED** [review]
Reapprove

**Review by mateusmm01 — APPROVED** [review]
Tested on ts4. Works as expected

**Pull request merged**